### DISCUSSION TITLE
v24.12.06

### DISCUSSION THREAD

**Discussion by davidrapan** [davidrapan/ha-solarman#297] [Announcements]
## What's Changed
* [refactor: Complete overhaul of integration initialization](https://github.com/davidrapan/ha-solarman/commit/71530d6150b8a182b406fee1333738b7472eba85)
* [fix: Workaround for modbus tcp responses w/o quantity](https://github.com/davidrapan/ha-solarman/commit/e68423ebdc2c822ab20325dd607a3f237127fbdd)
* [fix: Platform solarman does not generate unique IDs](https://github.com/davidrapan/ha-solarman/commit/02a7d8a666210367499fae49e66dbd497fd0a3eb)

**Full Changelog**: https://github.com/davidrapan/ha-solarman/compare/v24.12.04...v24.12.06

<hr /><em>This discussion was created from the release <a href='https://github.com/davidrapan/ha-solarman/releases/tag/v24.12.06'>v24.12.06</a>.</em>

**Comment by PV-Joe** [comment]
Just updated HA-Solarman to Version 24.12.06_4 for my Deye Hybrid 8K and everything's working fine with Home Assistant 2024.12.1

Thx for keeping this integration up to date 💜

**Reply by davidrapan** [reply]
It's interesting that the only device which almost never has any issues is our device, right? 😆 I wonder why that is..... 😆 very peculiar! 😆

**Comment by philmassie** [comment]
Hello
Just ran the update to 24.12.06 and seem to have broken a lot of things. All entity renames were tricky to fix and after trying to match them all manually, some things were reporting strange values (eg grid was reporting 220W while the inverter screen was reporting 20W, so I guess a multiplier somewhere).
Seems like `inverter_load_power` was also 10x 

Introduced a new entity `inverter_pv3_voltage` - I don't have any other pv3 entities.
Several old entities didn't seem to have new matches.

Got the migration failed error too.

I did click update without reading the changelog carefully, so that's on me :)

Ended up restoring a backup, and will watch for new updates.

Deye 5KW Hybrid

**Reply by davidrapan** [reply]
Hi,
* Grid and Load power sensors being * 10 is tracked:
  * https://github.com/davidrapan/ha-solarman/issues/281
  * https://github.com/davidrapan/ha-solarman/issues/298
  * And was reverted in [v24.12.06_4](https://github.com/davidrapan/ha-solarman/releases/tag/v24.12.06_4)
* If number of MPPTs was not detected correctly you can manually select it in the CONFIGURE >> Additional options
* Migration should be all good in [v24.12.06_4](https://github.com/davidrapan/ha-solarman/releases/tag/v24.12.06_4) and [v24.12.08](https://github.com/davidrapan/ha-solarman/releases/tag/v24.12.08)

Sh*t happens, that's what backups are for. 😉

**Reply by philmassie** [reply]
Thanks for all the hard work on this!

**Reply by philmassie** [reply]
Installed 24.12.08_5 this morning, and looking good, thanks again.
Also, my entities kept their same names for the most part (didn't change to `inverter_`)

Noticing the following `Power` entities are missing:

- External CT1 Power
- Grid L1 Power
- Load L1 Power
- Output L1 Power

And the following `Voltage` entity
- Load Voltage

But the following `Current` and `Voltage` are still there:

- External CT1 Current
- Grid L1 Current
- Load L1 Current
- Output L1 Current

- External CT1 Voltage
- Grid L1 Voltage
- Load L1 Voltage
- Output L1 Voltage

Other missing entities (all zeros in my case anyway) 
- Micro-inverter Power
- all L2

Thanks so much for all the work

**Reply by davidrapan** [reply]
Maybe you already figured it out or not but you can read more 'bout this behavior here: https://github.com/davidrapan/ha-solarman/issues/281#issuecomment-2531344178

**Reply by philmassie** [reply]
lol no I hadn't figured it out, thanks for that link :)

**Comment by davidrapan** [comment]
Could you both guys @PV-Joe and @philmassie also give me screenshots like this one? https://github.com/davidrapan/ha-solarman/issues/281#issuecomment-2520017035

**Reply by davidrapan** [reply]
I'd like to start track all the possible combinations for improving the autodetection feature. 😉

**Reply by philmassie** [reply]
Sure thing
![image](https://github.com/user-attachments/assets/899e2a80-763d-4e63-8a85-043fb2a83be4)

**Comment by PV-Joe** [comment]
> Could you both guys @PV-Joe and @philmassie also give me screenshots like this one? https://github.com/davidrapan/ha-solarman/issues/281#issuecomment-2520017035

Here you go:
![image](https://github.com/user-attachments/assets/25147045-75ae-4f1b-8380-ea241637c88d)

**Comment by philmassie** [comment]
with 24.12.08_5
Still seeing the following in the logs
`2024-12-08 12:33:59.254 ERROR (MainThread) [homeassistant.components.switch] Platform solarman does not generate unique IDs. ID solarman_2766832607_generator_switch already exists - ignoring switch.solarman_generator`

Should I log a bug or am I doing something dumb?

Thanks so much

**Reply by davidrapan** [reply]
Quick fix: [fix: Generator switch already exists - Deye Hybrid](https://github.com/davidrapan/ha-solarman/commit/cf3c8dc433255365a82d9b288a5701c87ec9322a), gonna have to look at it more later.
